### ISSUE TITLE
Parent item in structure linked to canvas

### ISSUE THREAD

**Issue by ch-sander** [ProjectMirador/mirador#3521]
It seems impossible for Mirador (and UV) to link a parent item in the structures ('Table of contents') to the canvas that is specified in the manifest.json but links will only be active for child items of the ultimate level. Put simple: If you have the 'Book 1' beginning at p. 10, and sub-items like 'Diagram 1/2/n', only these sub-items will be linked to the canvases, although there is a canvas also for the parent item in the json.

This is not how I as a reader would want to navigate and hierarchical PDF bookmarks work differently, too. Parent items in historical documents are often themself referers in the document and duplacting them seems awkward.

Can you fix this?

Example:

> "structures": [
    {
      "id": "https://dlib2.biblhertz.it/iiif/3/bdh0000089635/ranges",
      "type": "Range",
      "label": {
        "en": [
          "Table of Contents"
        ]
      },
      "items": [
        {
          "id": "https://dlib2.biblhertz.it/iiif/3/bdh0000089635/ranges/5",
          "type": "Range",
          "label": {
            "en": [
              "Book 1"
            ]
          },
          "items": [
            {
              "id": "https://dlib2.biblhertz.it/iiif/3/bdh0000089635/canvas/p0005",
              "type": "Canvas"
            },
            {
              "id": "https://dlib2.biblhertz.it/iiif/3/bdh0000089635/ranges/7",
              "type": "Range",
              "label": {
                "en": [
                  "Diagram 1"
                ]
              },
              "items": [
                {
                  "id": "https://dlib2.biblhertz.it/iiif/3/bdh0000089635/canvas/p0007",
                  "type": "Canvas"
                }
              ]
            }
          ]
        }

**Comment by brittnylapierre** [comment]
Hi all! Could someone explain the desired functionality more visually? This would help with design, thanks

**Comment by ch-sander** [comment]
Has been a while and the host's image server seems to be down for my own examples to test but if I remember correctly the issue is that only the child items (lowest level) link to a canvas, i.e. are clickable to load an image, while any parent structure item one unfolds the children but is not linked to a canvas in the UI.

I assume this is intended but it fails to address how (historical) documents work where some parent item, say a chapter, has children, say paragraphs, and users would want to naivigate to the canvas for a chapter to begin while the first paragraph is only e.g. two canvases later.

![grafik](https://github.com/user-attachments/assets/ef5bbc21-b9b8-47f1-9025-dca38f63d289)

---

# Index/Table of Contents UI Structure

- **Book Title (Clickable, leads to cover page canvas)**
  - **Chapter 1 (Not clickable, but should lead to the first canvas of the chapter)**
    - Paragraph 1.1 (Clickable, leads to its specific canvas)
    - Paragraph 1.2 (Clickable, leads to its specific canvas)
    - Paragraph 1.3 (Clickable, leads to its specific canvas)
  - **Chapter 2 (Not clickable, but should lead to the first canvas of the chapter)**
    - Paragraph 2.1 (Clickable, leads to its specific canvas)
    - Paragraph 2.2 (Clickable, leads to its specific canvas)

## Issue
- When users want to start reading from the beginning of a chapter, they cannot directly navigate to the chapter's starting canvas. Instead, they have to manually click through or find the first chapter's canvas.

## Proposed Solution
- Make each chapter header clickable and link it to the canvas where the chapter begins. This provides a more intuitive navigation experience, aligning with the traditional way users interact with physical documents.

**Comment by ch-sander** [comment]
maybe @giacomomarchioro remembers as well

**Comment by giacomomarchioro** [comment]
yes, for instance  when we click over the chapter in the table of content [like in this example](https://projectmirador.org/embed/?iiif-content=https://iiif.io/api/cookbook/recipe/0024-book-4-toc/manifest.json) the UI opens a dropdown, it does not set the current canvas to the chapter starting point, so the user has to click on the subsection. This is particularly annoying if the subsection start some pages after the Chapter starting point, the user that wants to start reading a chapter must first click on the subsection and then move up to the Chapter starting point. Other viewers don't behaves like this see for instance [theseus](https://theseusviewer.org/?iiif-content=https://iiif.io/api/cookbook/recipe/0024-book-4-toc/manifest.json), maybe @ch-sander could provide a link to a manifest that has a subsection starting some pages after the chapter starting point, the comparison between theseus and mirador could clarify the issue.

**Comment by ch-sander** [comment]
thanks @giacomomarchioro for clarification. Once our image server is back to work, I can point to an example [here](https://projectmirador.org/embed/?iiif-content=https://raw.githubusercontent.com/ch-sander/raramagnetica/main/iiif/garzoni.json). In my pseudocode in the initial comment, it's maybe best understood if you imagine a manifest structure that has chapters or books as top-level-structure and images or tables as sub-structure that rarely occur in the first page of a book/chapter.

**Comment by lutzhelm** [comment]
Linking this [discussion](https://github.com/ProjectMirador/mirador/pull/2916#issuecomment-605346974) (item 2) about the original implementation for reference.

**Comment by marlo-longley** [comment]
Claude analysis:

Root cause (src/components/SidebarIndexTableOfContents.jsx:84-94):
```
const selectTreeItem = (itemId) => {
  const node = deepFind(treeStructure, itemId);
  // Do not select if there are no canvases listed or it has children
  if (!node.data.getCanvasIds() || node.data.getCanvasIds().length === 0 || node.nodes.length > 0) {
    return;
  }
  ...
};
```
The node.nodes.length > 0 check blocks navigation for any Range with children, even though getStartCanvasId already knows how to read the spec's start/startCanvas property — it's just never called for parent Ranges.

Why it's not a simple revert: that guard was deliberately added in commit 8e429384 (2020) after PR #2916, where a user (jvine) reported that clicking a parent Range jumped to a confusing, arbitrary "first canvas in the subtree" when no explicit start was set. Naively removing the guard reintroduces that exact disorientation for Ranges lacking an explicit start/startCanvas.

What's changed since: Mirador now uses MUI-X SimpleTreeView, which already separates icon-click (expand) from label-click (select) — one of the two mitigations proposed back in 2020 exists for free. What's still undecided:
1. Should parent nodes navigate only when start/startCanvas is explicit (safe) vs. always fall back to first-contained-canvas (matches the ticket's ask, but risks the 2020 disorientation)?
2. What should Enter/Space do on a parent node (currently expand-only)?
3. Visual affordance — signaling a parente, distinct from the expand chevron.

-----

Designer work:

### 1. Interaction pattern for parent nodes
- Currently: click anywhere on a parent Range row → expand/collapse only.
- Proposed: click the label → navigate to that Range's start canvas; click the icon/chevron → expand/collapse. (Mirador's MUI-X tree already fires separate events for icon vs. label, so this split is feasible — but it needs to be a deliberate, stated interaction, not just "whatever the click handler happens to do.")
- Designer needs to decide: is single-click-on-label-navigates the right mental model, or is there a clearer alternative (e.g., a small "jump to page" affordance separate from the label)?

### 2. Visual affordance
- Right now a parent row looks identical whether it navigates or not. If label-click starts doing something, users need a visual cue that it's clickable (hover state, cursor, underline, icon) distinct from the expand chevron — otherwise it's an invisible feature.
- Also needs a state for parent Ranges that can't navigate (no start/startCanvas and no sensible fallback) — should those look visually different (non-interactive label) from ones that can?

### 3. Fallback behavior when there's no explicit start/startCanvas
- Spec-compliant case is easy: Range has start → clicking label goes there.
- Ambiguous case: Range has no start, only children. Options:
  - (a) Label is not clickable at all — stays expand-only (safest, avoids 2020's "jumps to a random blank page" complaint).
  - (b) Label falls back to navigating to the first canvas in the subtree in document order (matches what some users expect, but risks the exact disorientation the 2020 fix was written to prevent).
  - Designer needs to pick one, or design a way to signal to the user which case they're in.

### 4. Keyboard behavior
- Enter/Space on a focused parent node currently just expands. If label-click now navigates, does Enter/Space do both (expand + navigate) in one keystroke, or does keyboard navigation need a separate way to trigger "go to canvas" vs. "expand"? This is a real accessibility decision, not just a mouse-interaction afterthought.

### 5. Edge cases worth sketching
- A parent Range whose own canvas is the same as its first child's canvas (no-op navigation — fine).
- A parent Range whose canvas is far from its children's canvases (the "Chapter 1 starts at p.10, Paragraph 1.1 is at p.14" case from the ticket) — this is the main case users are asking for.
- Toggling a parent closed after having navigated into a child — should collapsing also change the canvas back to the parent's, or leave the current canvas alone? (This was literally part of jvine's original 2020 complaint.)
